src/pages/billing/NewBillingProjectWizard/AzureBillingProjectWizard/AzureSubscriptionStep.ts

```diff
@@ -0,0 +1,158 @@
+import { validate } from 'validate.js'
+
+
+type AzureSubscriptionStepProps = {
+  isActive: boolean
+  subscriptionId?: string // undefined indicates the value hasn't been changed by the user yet
+  onSubscriptionIdChanged: (string, boolean) => void
+  managedApp?: AzureManagedAppCoordinates
+  onManagedAppSelected: (AzureManagedAppCoordinates) => void
+  inputDebounce?: number
+}
+
+const managedAppsToOptions = (apps: AzureManagedAppCoordinates[]) => _.map(application => {
+  return {
+    value: application,
+    label: !!application.region ?
+      `${application.applicationDeploymentName} (${application.region})` :
+      application.applicationDeploymentName
+  }
+}, apps)
+
+// @ts-ignore
+validate.validators.type.types.uuid = value => validateUuid(value)
+// @ts-ignore
+validate.validators.type.messages.uuid = 'must be a UUID'
+
+
+export const AzureSubscriptionStep = ({ isActive, subscriptionId, inputDebounce = 0, ...props }: AzureSubscriptionStepProps) => {
+  const [subscriptionIdError, setSubscriptionIdError] = useState<ReactNode>()
+  const [managedApps, setManagedApps] = useLoadedData<AzureManagedAppCoordinates[]>({
+    onError: state => {
+      // We can't rely on the formatting of the error, so show a generic message but include the error in the console for debugging purposes.
+      if (state.error instanceof Response) {
+        state.error.text().then(console.error)
+      } else {
+        console.error(state.error)
+      }
+      setSubscriptionIdError(h(NoManagedApps))
+    }
+  })
+  const subscriptionIdInput = useRef<HTMLInputElement>()
+  const subscriptionInputId = useUniqueId()
+  const appSelectId = useUniqueId()
+  const signal = useCancellation()
+
+  const getSubscriptionIdErrors = subscriptionId => subscriptionId !== undefined && validate({ subscriptionId }, { subscriptionId: { type: 'uuid' } })
+
+  useEffect(() => {
+    // setTimeout necessary because of UIE-73.
+    setTimeout(() => subscriptionIdInput.current?.focus(), 0)
+  }, [])
+
+  const subscriptionIdChanged = v => props.onSubscriptionIdChanged(v, !!getSubscriptionIdErrors(v))
+
+  // this is done in a useEffect instead of just calculating from props to debounce the input validation
+  useEffect(() => {
+    const timeoutId = setTimeout(() => {
+      const subscriptionIdErrors = getSubscriptionIdErrors(subscriptionId)?.subscriptionId
+      if (!!subscriptionIdErrors) {
+        setSubscriptionIdError(summarizeErrors(subscriptionIdErrors))
+      } else if (!!subscriptionId) {
+        setManagedApps(async () => {
+          setSubscriptionIdError(undefined)
+          const response = await Ajax(signal).Billing.listAzureManagedApplications(subscriptionId, false)
+          const managedApps = response.managedApps
+          if (managedApps.length === 0) {
+            setSubscriptionIdError(h(NoManagedApps))
+          }
+          return managedApps
+        })
+      }
+    }, inputDebounce)
+    return () => clearTimeout(timeoutId)
```

Comment:
@blakery let's remove the debouncing on this field to simplify the code. It will also remove the need to deal with `setManagedApps` as a dependency.

Users are most likely going to paste into this field, but even if they type they won't see the error message changing with every keystroke as the UUID will be invalid up until they type the last digit.

Reply:
Ugh, looks like `setManagedApps` dependency issue was introduced by the code changes for `useLoadedData`. I don't like ignoring the warning-- @msilva-broad is there another option to avoid having to ignore dependencies?

https://github.com/DataBiosphere/terra-ui/pull/3790/files/03a438e91d59e905b89df07ad687948e8a3b9844#diff-55e8cfc1bf896fdf1117687f5a3aa9a8c761f88cc34c289c45ecba66f3f694aeR100

Reply:
@msilva-broad we no longer need the `useEffect` callback, so the need to disable the eslint warning is moot.

But I'm still curious about the general issue-- is there a way to update React/eslint so that it treats `useLoadedData` as a dependency that it wan't warn about (in the same way that it doesn't warn about `useState`)?